crates/consensus/auto-seal/src/task.rs

```diff
@@ -131,7 +132,11 @@ where
                     let substate = SubState::new(State::new(client.latest().unwrap()));
                     let mut executor = Executor::new(chain_spec, substate);
 
-                    match storage.build_and_execute(transactions.clone(), &mut executor) {
+                    match storage.build_and_execute(
+                        transactions.clone(),
+                        &mut executor,
+                        chain_spec2,
+                    ) {
```

Comment:
no need to clone the chain spec twice outside of the closure, you should be able to do it inside
```rust
let mut executor = Executor::new(Arc::clone(&chain_spec), substate);

match storage.build_and_execute(
    transactions.clone(),
    &mut executor,
    chain_spec,
 ) {
```

Reply:
done